glib-macros/src/properties.rs

```diff
@@ -0,0 +1,527 @@
+        });
+        let setter = p.set.as_ref().map(|mfn| {
+            let body = match (p.member.as_ref(), mfn) {
+                (None, MaybeCustomFn::Default) => quote!(
+                     self.imp().#field_ident.set(move |x| *x = value)
+                ),
+                (Some(member), MaybeCustomFn::Default) => quote!(
+                     self.imp().#field_ident.set(move |x| x.#member = value)
+                ),
+                (_, MaybeCustomFn::Custom(custom_fn)) => {
+                    let custom_fn = change_self_type(custom_fn, &imp_type_ident.to_string());
+                    quote!((#custom_fn)(&self.imp(), value))
+                }
+            };
+            let fn_prototype = setter_prototype(&ident, ty);
+            quote!(#fn_prototype { #body })
+        });
+        quote!(
+            #getter
+            #setter
+        )
+    });
+    quote!(#(#defs)*)
+}
+
+fn expand_connect_prop_notify(p: &PropDesc) -> TokenStream2 {
+    let name = &p.name;
+    let fn_ident = format_ident!("connect_{}_notify", name_to_ident(name));
+    quote!(fn #fn_ident<F: Fn(&Self) + 'static>(&self, f: F) -> glib::SignalHandlerId)
+}
+fn expand_connect_prop_notify_def(props: &[PropDesc]) -> TokenStream2 {
+    let connection_fns = props.iter().map(expand_connect_prop_notify);
+    quote!(#(#connection_fns;)*)
+}
+fn expand_connect_prop_notify_impl(props: &[PropDesc]) -> TokenStream2 {
+    let connection_fns = props.iter().map(|p| {
+        let name = &p.name;
+        let fn_prototype = expand_connect_prop_notify(p);
+        quote!(#fn_prototype {
+            self.connect_notify_local(Some(#name), move |this, _| {
+                f(this)
+            })
+        })
+    });
+    quote!(#(#connection_fns)*)
+}
+
+pub fn impl_derive_props(input: PropsMacroInput) -> TokenStream {
+    let struct_ident = &input.ident;
+    let struct_ident_ext = format_ident!("{}Ext", &input.ident);
+    let wrapper_type = quote!(<#struct_ident as glib::subclass::types::ObjectSubclass>::Type);
+    let fn_properties = expand_properties_fn(&input.props);
+    let fn_property = expand_property_fn(&input.props);
+    let fn_set_property = expand_set_property_fn(&input.props);
+    let getset_properties_def = expand_getset_properties_def(&input.props);
+    let getset_properties_impl = expand_getset_properties_impl(struct_ident, &input.props);
+    let connect_prop_notify_def = expand_connect_prop_notify_def(&input.props);
+    let connect_prop_notify_impl = expand_connect_prop_notify_impl(&input.props);
+    let expanded = quote! {
+        use glib::{PropertyRead, PropertyWrite};
+
+        impl glib::subclass::object::DerivedObjectProperties for #struct_ident {
+            #fn_properties
+            #fn_property
+            #fn_set_property
+        }
+
+        pub trait #struct_ident_ext {
+            #getset_properties_def
+            #connect_prop_notify_def
+        }
+        impl #struct_ident_ext for #wrapper_type {
```

Comment:
```suggestion
        impl<T: #crate_ident::IsA<#wrapper_type>> #struct_ident_ext for T {
```

It needs to be this to work with inherited types, too